### PR TITLE
Added Python 3.13 to CI and Descriptions

### PR DESCRIPTION
**Pull request by kc611**
As titled, this PR adds Python 3.13 as a part of CI which is planned to be used to make a Python 3.13 `llvmlite` build.

Note: Currently Python 3.13 is available in the [`ad-testing`](https://anaconda.org/ad-testing/python) channel within conda and it will be some time before it makes it to the main channel.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -66,6 +74,10 @@ jobs:
         PYTHON: '3.12'
         CONDA_ENV: cienv
         WHEEL: 'yes'
+      py312_wheel:
```
Comment (sklam) [inline azure-pipelines.yml:77]: Is this suppose to be for 3.13?
it's failing azure: https://dev.azure.com/numba/numba/_build/results?buildId=17733&view=results

**Comment by stuartarchibald** [comment]
I think Python 3.9 should be dropped to match Numba's support (only 4 Python versions officially supported and tested)? Should this suggestion be accepted, would the change go into this PR or into a subsequent PR?

**Comment by cclauss** [comment]
Is there a remaining todo on this?

**Comment by kc611** [comment]
> Is there a remaining todo on this?

Yep, we need to bump the minimum supported Python version to 3.10 according to @stuartarchibald's comment above.

**Comment by cclauss** [comment]
> would the change go into this PR or into a subsequent PR?

**Comment by gmarkall** [comment]
Requires a merge from `main` to resolve the RTD issue.

**Review by sklam — APPROVED** [review]

**Comment by jakirkham** [comment]
Cross-linking to Python 3.13 support issue

xref: https://github.com/numba/llvmlite/issues/1084

**Pull request merged**